### ISSUE TITLE
MW 1.25 / Local test failures

### ISSUE THREAD

**Issue by JeroenDeDauw** [SemanticMediaWiki/SemanticMediaWiki#724]
Local wiki with MediaWiki master and SMW master.

Getting these failures when in run `php phpunit.php -c ../../extensions/SemanticMediaWiki/` from the `tests/phpunit` directory: http://pastebin.com/c4cQDMep

**Comment by JeroenDeDauw** [comment]
`php phpunit.php -c ../../extensions/SemanticMediaWiki/ --testsuite=semantic-mediawiki-unit` all passes \o/

**Comment by mwjames** [comment]
> 1) SMW\Tests\Integration\MediaWiki\Hooks\FileUploadIntegrationTest::testFileUploadForDummyTextFile
fopen(/var/www/phase3/images/lockdir/mk1j29pngfx427h8i9zbvu7ynv0381z.lock): failed to open stream: Permission denied

Permission issue, Travis works.


> 3) SMW\Tests\Integration\MediaWiki\Import\CategoryInstanceAndCategoryHierarchyTest::testCategoryHierarchies
Use of WikiImporter::__construct without a Config instance was deprecated in MediaWiki 1.25. [Called from SMW\Tests\Utils\Runners\XmlImportRunner::run in /var/www/phase3/extensions/SemanticMediaWiki/tests/phpunit/Utils/Runners/XmlImportRunner.php at line 67]

Some WMF funny bone changed the interface in 1.25 and I didn't bother to run around to fix those issues.

https://github.com/SemanticMediaWiki/SemanticMediaWiki/commit/7c6c1abcb09a4e89e2c632cffe97f081193b1a3d#diff-5396a58b13a8d9dd9eafea671f24bfa5

> 1) SMW\Tests\Integration\MediaWiki\Hooks\TitleMoveCompleteIntegrationTest::testPageMoveWithCreationOfRedirectTarget
Failed asserting that null is not null.

TitleMove issues (see #607). No one from the WMF bothers to even look at https://phabricator.wikimedia.org/T74870 therefore I don't really care to change our tests that worked until `master@5cc1f1d` but not after.

**Comment by JeroenDeDauw** [comment]
> Permission issue, Travis works.

Any idea how to fix this? Seems like that when I run the test in isolation, I get a different error: http://pastebin.com/cu96vkpy

**Comment by mwjames** [comment]
> Any idea how to fix this? Seems like that when I run the test in isolation, I get a different error: http://pastebin.com/cu96vkpy

> Query: SELECT  *  FROM `sunittest_smw_fpt_media`   WHERE s_id = '52'  
Function: SMWSQLStore3Writers::getCurrentPropertyTableContents
Error: 1146 Table 'mw_dev.sunittest_smw_fpt_media' doesn't exist (localhost)

The unit test table cloner expects that all tables are available before cloning by the time the test is run (never executed this test in isolation). To have the `smw_fpt_media` table available settings like `'smwgPageSpecialProperties' => array( '_MEDIA', '_MIME' )` are necessary to have the `setupStore` to create the tables but the integration tests normally run the `setupStore` + `tableCloner` at the start.

https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Integration/MediaWiki/Hooks/FileUploadIntegrationTest.php#L55

The temp file is created by https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/tests/phpunit/Utils/Fixtures/File/DummyFileCreator.php#L95.

**Comment by mwjames** [comment]
As a side note for the lazy worker among us you can just use the `composer phpunit` as in `composer phpunit -- --filter FileUploadIntegrationTest` and running it doesn't fail:
```
$ composer phpunit -- --filter FileUploadIntegrationTest

MediaWiki 1.23.3 root vendor autoloader ...

SMWSQLStore3 (mysql) ...

PHPUnit 4.3.1 by Sebastian Bergmann.

Configuration read from ...\mw-123\extensions\SemanticMediaWiki\phpunit.xml.
...

Time: 1.36 minutes, Memory: 68.75Mb

OK (3 tests, 32 assertions)
```

**Comment by JeroenDeDauw** [comment]
Cool! I had not seen that script yet :)

**Comment by mwjames** [comment]
> fopen(/var/www/phase3/images/lockdir/mk1j29pngfx427h8i9zbvu7ynv0381z.lock): failed to open stream: Permission denied

The image folder requires to be writeable (777) (see https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/750#issuecomment-71432889).

**Comment by JeroenDeDauw** [comment]
Works fine with MW master on my laptop

**Comment by mwjames** [comment]
> Works fine with MW master on my laptop

You should see issues with MW master (1.25+) when running the integration tests.

**Comment by mwjames** [comment]
Fixed with c2577df6da
